controllers/filesController.js

```diff
@@ -1,194 +1,214 @@
-    const ur = await QRCode.toDataURL(`${file.fileId}`);
-    const qr = ur.substring(22);
-    res.status(200).json({
-        status: "success",
-        data:{
-            history,
-            file,
-            qr
-        }
-    })
+exports.setFileHistory = asyncWrapper(async (req, res, next) => {
+  const fileId = req.params.fileId;
+  const { info } = req.body;
+  let file = null;
+  try {
+    file = await Files.findOne({ fileId });
+  } catch (e) {
+    return res.status(400).json({
+      status: "fail",
+      data: {
+        message: "invalid fileId, no file belong to this fileId",
+      },
+    });
+  }
+  const [lastSpot] = await FileHistory.aggregate([
+    {
+      $group: {
+        _id: null,
+        recent: {
+          $max: {
+            date: "$reachedAt",
+            userId: "$userId",
+            _id: "$_id",
+          },
+        },
+      },
+    },
+  ]);
+  if (lastSpot.recent.userId === req.user._id) {
+    return res.status(200).json({
+      status: "success",
+      data: {
+        file,
+        message: "you have alredy scanned the file.",
+      },
+    });
+  }
+  const spot = {
+    fileId,
+    userId: req.user._id,
+    info: info.length ? info : req.user.department ? req.user.department : "",
+    reachedAt: req.requestTime,
+  };
+  await FileHistory.create(spot);
+  const history = await FileHistory.find({ fileId: fileId })
+    .populate("userId")
+    .sort("reachedAt");
+  const ur = await QRCode.toDataURL(`${file.fileId}`);
+  const qr = ur.substring(22);
+  res.status(200).json({
+    status: "success",
+    data: {
+      history,
+      file,
+      qr,
+    },
+  });
 });
 
-exports.getRecentFiles = asyncWrapper(async(req, res, next)=>{
-    const recentFiles = await FileHistory.aggregate([
-        {
-            $lookup:{
-                from: "files",
-                foreignField:"fileId",
-                localField: "fileId",
-                as: "file"
-            } 
-        },
-        {
-            $match:{ "file.owner": mongoose.Types.ObjectId(req.user._id) }
-        },
-        {
-            $unwind: "$file"
-        },
-        {
-            $group: {
-                _id: "$file.fileId",
-                reachedAt: { $max :"$reachedAt"},
-                fileName: { $first: "$file.fileName"},
-                description: { $first: "$file.description"}
-            }
-        },
-        {
-            $sort: {
-                "reachedAt": -1
-            }
-        },
-        {
-            $limit: 5
-        }
-    ])
-    res.status(200).json({
-        status: "success",
-        data:{
-            recentFiles
-        }
-    })
-})
\ No newline at end of file
+//delete fileHistory
+exports.deleteFileHistory = asyncWrapper(async (req, res, next) => {
+  const { fileId } = req.params;
+  const file= await Files.findOne({fileId:fileId});
+  if (!file) {
+    return res.status(400).json({
+      status: "fail",
+      message: "file not found",
+    });
+  }
```

Comment:
Add one more condition to check if the request is made by file Owner or not if yes than delete file else not.